<!-- 
    ROS navigation stack withOUT velocity smoother and safety (reactive) controller
-->
<launch>
  
  <arg name="odom_frame_id"   default="odom"/>
  <arg name="base_frame_id"   default="base_footprint"/>
  <arg name="global_frame_id" default="map"/>
  <arg name="odom_topic" default="odom" />
  <arg name="laser_topic" default="scan" />
  <arg name="custom_param_file" default="$(find turtlebot_navigation)/param/dummy.yaml"/>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base_multimaster" output="screen">
    <rosparam file="$(find vln_agent)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find vln_agent)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find vln_agent)/param/local_costmap_params.yaml" command="load" />   
    <rosparam file="$(find vln_agent)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find vln_agent)/param/dwa_local_planner_params.yaml" command="load" />
    <rosparam file="$(find vln_agent)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find vln_agent)/param/global_planner_params.yaml" command="load" />
    <rosparam file="$(find vln_agent)/param/navfn_global_planner_params.yaml" command="load" />
    <!-- external params file that could be loaded into the move_base namespace -->
    <rosparam file="$(arg custom_param_file)" command="load" />
    
    <!-- reset frame_id parameters using user input data -->
    <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
    <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="local_costmap/global_frame" value="$(arg odom_frame_id)"/>
    <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="DWAPlannerROS/global_frame_id" value="$(arg odom_frame_id)"/>

    <remap from="cmd_vel" to="navigation_velocity_smoother/raw_cmd_vel"/>

    <!-- remap a bunch of stuff to avoid clashes with the other move_base -->
    <remap from="move_base_simple/goal" to="move_base_multimaster_simple/goal"/>
    <remap from="move_base/feedback" to="move_base_multimaster/feedback"/>
    <remap from="move_base/status" to="move_base_multimaster/status"/>
    <remap from="move_base/result" to="move_base_multimaster/result"/>
    <remap from="move_base/goal" to="move_base_multimaster/goal"/>
    <remap from="move_base/cancel" to="move_base_multimaster/cancel"/>

    <remap from="odom" to="$(arg odom_topic)"/>
    <remap from="scan" to="$(arg laser_topic)"/>
  </node>
</launch>
